File: python/QEasyWidgets/Common/Signals.py
from PySide6.QtCore import QObject, Signal

##############################################################################################################################

class CustomSignals_ComponentsCustomizer(QObject):
    '''
    Set up signals for components
    '''
    # Set theme
    setTheme = Signal(str)

    # Set language
    setLanguage = Signal(str)


componentsSignals = CustomSignals_ComponentsCustomizer()

##############################################################################################################################

File: c++/QEasyWidgets/Common/Signals.cpp
#include "Signals.h"
#include "Translator.h"
#include "Theme.h"

namespace QEW {

CustomSignals::CustomSignals(QObject *parent)
    : QObject(parent)
{
    // Connect language change signal
    connect(this, &CustomSignals::setLanguage, this, [](const QString &lang) {
        updateLanguage(lang);
    });
    
    // Connect theme change signal
    connect(this, &CustomSignals::setTheme, this, [](const QString &themeStr) {
        if (themeStr == "light" || themeStr == ThemeString(LIGHT)) {
            setCurrentTheme(LIGHT);
        } else if (themeStr == "dark" || themeStr == ThemeString(DARK)) {
            setCurrentTheme(DARK);
        } else if (themeStr == "auto" || themeStr == ThemeString(AUTO)) {
            setCurrentTheme(detectSystemTheme());
        }
    });
}

CustomSignals *componentsSignals = new CustomSignals();

} // namespace QEW